### PR TITLE
build.sh: keep proactive support patch idempotence, but fail fast if patch fails

### PR DESCRIPTION
**Pull request by granders**
Previously when patching kafka with proactive support, the build would proceed even if the proactive support patch failed.

This updates so that proactive support patch is still idempotent, but fails the build if the patch fails to apply.

### CODE REVIEW

**Comment by granders** [comment]
@ewencp @Ishiihara Mind taking a quick look?

File: build.sh [hunk 1]
```diff
@@ -82,19 +82,39 @@ function check_jdk_version {
...
+function apply_patch_file {
+    local directory=$1
+    local patch_file=$2
+
+    if [[ ! -d "$directory" ]];then
+        echo "Expected directory ${directory} to exist, but it does not"
         exit 1
     fi 
-    cd $PROJECTS_DIR/kafka
+    cd "$directory"
```
Comment (ewencp) [inline build.sh]: Hmm, it looks like using `cd` worked before, but this is even more risky in a generic method. `pushd` `popd` instead?

**Comment by ewencp** [comment]
Minor comment, but with the caveat that I have not run this myself it LGTM

**Comment by granders** [comment]
@ewencp Thanks, we'll just hope I didn't break it too badly.

**Pull request merged**